### PR TITLE
gci/feeds: Create RSS feed of tasks

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Creates RSS feed for tasks. Task descriptions are rendered HTML from markdown, along with external URL, if any. Link points to GCI task link.

Closes https://github.com/coala/community/issues/57

### CODE REVIEW

File: community/settings.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['127.0.0.1', 'localhost', 'testserver']
```
Comment (hemangsk) [inline community/settings.py]: Unrelated change?

Comment (nalinbhardwaj) [reply community/settings.py]: No, the build will fail because the syndication feed framework needs `testserver` to be allowed for `django_distill` builds.

File: community/urls.py [hunk 2]
```diff
@@ -37,6 +38,12 @@ def get_index():
         distill_func=get_index,
         distill_file='activity/index.html',
     ),
+    distill_url(
+        r'gci/rss/', gci_rss(),
+        name='gci-rss',
+        distill_func=get_index,
+        distill_file='gci/rss/index.xml',
```
Comment (jayvdb) [inline community/urls.py]: `gci/tasks/rss.xml` ?

File: community/settings.py (hunk 1 above)
Comment (hemangsk) [reply community/settings.py]: Okay 👍

File: .ci/build.sh [hunk 3]
```diff
@@ -9,7 +9,7 @@ if [[ -n "$GCI_TOKEN" ]]; then
   python manage.py cleanse_gci_task_data private _site
   rm -rf private/
 else
-  python manage.py fetch_old_gci_task_data _site || true
+  python manage.py fetch_old_gci_task_data private || true
```
Comment (jayvdb) [inline .ci/build.sh]: oh, dont touch this :P

File: community/urls.py (hunk 2 above)
Comment (nalinbhardwaj) [reply community/urls.py]: `index.xml` is better since then you don't have to type the whole thing, how about `gci/tasks-rss/` or something then?

File: gci/feeds.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class LatestTasksFeed(Feed):
+    title = 'GCI tasks feed'
+    link = get_deploy_url() + '/gci/rss/'
+    description = 'GCI tasks ordered by modification time.'
+
+    def items(self):
+        yaml = YAML()
+        with open(os.path.join('private', 'tasks.yaml')) as f:
```
Comment (jayvdb) [inline gci/feeds.py]: you need to load `_site/tasks.yaml`.  `private/tasks.yaml` is data which is private and can not be published.

File: .ci/build.sh (hunk 3 above)
Comment (nalinbhardwaj) [reply .ci/build.sh]: Builds won't work without API token then?

File: gci/feeds.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+        return res
+
+    def item_title(self, item):
+        return item['name']
+
+    def item_description(self, item):
+        desc = item['description']
+        if item['external_url']:
+            desc += '\n\nExternal URL: [' + item['external_url'] + \
```
Comment (jayvdb) [inline gci/feeds.py]: all this string appending is a bit ugly.  Could you use `%s` or `{url}`

File: community/urls.py [hunk 6]
```diff
@@ -37,6 +38,12 @@ def get_index():
         distill_func=get_index,
         distill_file='activity/index.html',
     ),
+    distill_url(
+        r'gci/rss/', gci_rss(),
```
Comment (jayvdb) [inline community/urls.py]: this and the next line also need to be updated.

File: community/urls.py [hunk 7]
```diff
@@ -8,6 +8,7 @@
 from django.views.generic import TemplateView
 
 from gci.views import index as gci_index
+from gci.feeds import LatestTasksFeed as gci_rss
```
Comment (jayvdb) [inline community/urls.py]: gci_tasks_rss
